Solution1/WeatherApi/Startup.cs

```diff
@@ -0,0 +1,63 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using WeatherApi.Configuration;
+using WeatherApi.Extensions;
+
+namespace WeatherApi
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var configSection = Configuration.GetSection("AppParams");
+            services.Configure<Config>(configSection);
```

Comment:
give meaningful name to  Config - the same as AppParams, and use nameof

Reply:
+